### PR TITLE
Feature/korge.optimizations

### PR DESCRIPTION
**Pull request by soywiz**
Profiling 10K sprite sample. I did these optimizations.

Plus one typo in the sample itself that affected the performance a lot:
https://github.com/korlibs/korge-samples/commit/f3149874791cc47b7399ef1f896288b8f61972ae

### CODE REVIEW

File: korge/src/commonMain/kotlin/com/soywiz/korge/render/BatchBuilder2D.kt [hunk 1]
```diff
@@ -662,15 +662,61 @@ class TexturedVertexArray(var vcount: Int, val indices: IntArray, var isize: Int
...
+        i32[pos + 4] = colMul.value
+        i32[pos + 5] = colAdd
+    }
+
     /**
      * Sets a textured quad at vertice [index] with the region defined by [x],[y] [width]x[height] and the [matrix],
      * using the texture coords defined by [BmpSlice] and color transforms [colMul] and [colAdd]
      */
+    @OptIn(KorgeInternal::class)
 	fun quad(index: Int, x: Double, y: Double, width: Double, height: Double, matrix: Matrix, bmp: BmpSlice, colMul: RGBA, colAdd: Int) {
```
Comment (soywiz) [inline korge/src/commonMain/kotlin/com/soywiz/korge/render/BatchBuilder2D.kt:679]: In KorGE 2.0 we should consider using Float for Matrix, or creating a Matrixf using floats

**Comment by emign** [comment]
> Profiling 10K sprite sample. I did these optimizations.
> 
> 
> 
> Plus one typo in the sample itself that affected the performance a lot:
> 
> https://github.com/korlibs/korge-samples/commit/f3149874791cc47b7399ef1f896288b8f61972ae

I’m on mobile. What’s the typo ? :)

**Comment by soywiz** [comment]
@emign You were creating all the SpriteAnimations on all the frames for all the sprites :) instead of reusing the animation

**Comment by soywiz** [comment]
Further optimizations will require some adjustments in other libraries like Klock: https://github.com/korlibs/klock/pull/113
It seems that Kotlin/Native is not removing boxings for inline + Number

**Comment by soywiz** [comment]
This commit: https://github.com/korlibs/korge-plugins/commit/ca2fb26475d61ce7e7e4662b21499136a9c4b837
Fixes the performance issue on 10K sprites sample on JS to be similar to the JVM. The `is ` patch was not applied anymore due to changes in the output of the kotlin.js runtime.

https://github.com/JetBrains/kotlin/pull/1817

**Pull request merged**